### DISCUSSION TITLE
Eto forms

### DISCUSSION THREAD

**Discussion by reubene** [axx0/Civ2-clone#50] [Ideas]
Can I ask why Eto forms was chosen? 

Was there any consideration to other cross platform C#  forms libraries?

I've been trying to get the mac version into a playable state and the problems I'll need to solve are not simple

**Comment by axx0** [comment]
Quite simply I was looking for Winforms equivalent with support for other platforms and this seemed like a good choice to try out (I was working on winforms-only release initially and since this supports it natively it was an easy choice). While it works for Windows there are issues for Linux I haven't addressed yet and I'm not sure about the nature of issues you're having on Mac since I don't own it (tbh I haven't received any feedback from anyone either). There's a [new](https://github.com/picoe/Eto/releases/tag/2.7.0) release which has some work on mac but I'm not sure if it would help.

Having a library which requires minimal work on widgets and cross-platforming while enabling us to at least partially retain the look and feel of the old game would be ideal but I haven't invested nearly enough energy into researching it. Do you have any ideas/suggestions?

**Comment by reubene** [comment]
It's very broken on Mac, it works but there are issues with compositing bitmaps so you end up with weird triangles everywhere. I did manage to get the buttons etc working though. 

I don't really have any other ideas, I've been out of the C# desktop app space for a long time, not really sure what the state of the art is.

**Reply by axx0** [reply]
Can you report the issue on Eto github page? If there is no solution we'll have to move to another library.

**Reply by reubene** [reply]
I don't know if it's an eto issue or an artifact of the way we're using, I'll look into if further before I raise an issue.

**Comment by axx0** [comment]
The newest thing seems to be [WinUI in combination with Uno](https://platform.uno/how-it-works/) for cross-platforming. I've been playing around with it for the last couple of days and let's just say porting over there won't be simple.

**Comment by axx0** [comment]
I've been testing various desktop environments. [Modern forms ](https://github.com/modern-forms/Modern.Forms) is what we need, but sadly it's very early in life and slow in development.

So Avalonia seems to be the way to go. It requires some knowledge of XAML but it shouldn't be a problem really. It would make stuff _much_ easier compared to Eto. No need to mess with custom controls etc. I'll try to port a part of this project so we can test it on different platforms.
